### PR TITLE
2/add logger

### PR DESCRIPTION
**Pull request by anxolin**
Part of #2 

Create a basic logger utility

It uses `https://www.npmjs.com/package/debug` and make it easy to create different log levels that can be filtered out later.

See usage in `bot.ts`

image

### CODE REVIEW

File: src/util/Logger.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      } else if (logLevel === 'INFO') {
+        consoleFn = 'info'
+      } else if (logLevel === 'WARN') {
+        consoleFn = 'warn'
+      }
+
+      if (consoleFn) {
+        // Set the console logger function (to use STDOUT)
+        // Note that by default is console.error
+        logger.log = console[consoleFn].bind(console)
```
Comment (Velenir) [inline src/util/Logger.ts]: Don't need `bind` console methods are static

File: src/bot.ts [hunk 2]
```diff
@@ -1,11 +1,10 @@
 import { strict as assert } from 'assert'
 import TelegramBot, { Message } from 'node-telegram-bot-api'
-
-const debug = require('debug')('DEBUG-bot')
-const info = require('debug')('INFO-bot')
+import Logger from './util/Logger'
 
 require('dotenv').config()
 
+const log = new Logger('bot')
```
Comment (Velenir) [inline src/bot.ts]: do you need to instantiate Logger every time, potentially in every file if there will be more. Wouldn't a singleton or static methods be better

File: src/util/Logger.ts (hunk 1 above)
Comment (anxolin) [reply src/util/Logger.ts]: Got it from https://www.npmjs.com/package/debug#output-streams

File: src/util/Logger.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  public error (formatter: any, ...args: any[]) {
+    this._log('ERROR', formatter, ...args)
+  }
+
+  public _log (level: string, formatter: any, ...args: any[]) {
+    const logger = this._getLogger(level, this._namespace)
+    if (args.length > 0) {
+      logger(formatter, ...args)
+    } else {
+      logger(formatter)
```
[the remark this answers is not in the document]
Comment (anxolin) [reply src/util/Logger.ts]: I'll do this in the PR at the bottom of the:
<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia1.giphy.com%2Fmedia%2F10olZ3QsdfXeSY%2Fgiphy.gif&token=waterfall" data-gitmeme-token="waterfall"></a>

File: src/util/Logger.ts (hunk 1 above)
Comment (Velenir) [reply src/util/Logger.ts]: They also don't need it then :tomato:

Comment (anxolin) [reply src/util/Logger.ts]: Are you sure?? it really depends on they use it.
I think it doesn't harm, and I don't want to dive in their lib, os I would stick to their documentation

**Pull request merged**